### PR TITLE
feat(schema): Better utility for JSON parsing

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

More convenient way to parse (and validate) JSON strings

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — DISMISSED** [review]
nice

**Comment by viceice** [comment]
maybe also provide a Json5 util?

**Comment by zharinov** [comment]
Good idea, and YAML maybe too

File: lib/util/schema-utils.ts [hunk 1]
```diff
@@ -91,26 +91,20 @@ export function looseValue<T, U extends z.ZodTypeDef, V>(
...
 }
 
-export function parseJson<
-  T = unknown,
-  Schema extends z.ZodType<T> = z.ZodType<T>
->(input: string, schema: Schema): z.infer<Schema> {
-  const parsed = JSON.parse(input);
-  return schema.parse(parsed);
-}
+type JsonData =
```
Comment (JamieMagee) [inline lib/util/schema-utils.ts:94]: [`type-fest`][1] has utility types for JSON objects that might be easier to use than maintaining our own.

[1]: https://github.com/sindresorhus/type-fest#json

**Review by JamieMagee — DISMISSED** [review]

**Review by JamieMagee — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**